sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc

```diff
@@ -37,155 +37,168 @@
+    //, fFlatEngine(art::ServiceHandle<rndm::NuRandomService>()->createEngine(*this, "HepJamesRandom", "Gen", p, "Seed"))
+    // Initialize member data here.
 {
 
-    //
-    // Random engine initialization
-    //
-    fFlatRandom = std::make_unique<CLHEP::RandFlat>(fFlatEngine,0,1);
-    fNormalRandom = std::make_unique<CLHEP::RandGauss>(fFlatEngine);
-
-    produces< std::vector<simb::MCTruth>   >();
-    produces< sumdata::RunData, art::InRun >();
-
-    _debug = p.get<unsigned short>("DebugMode",0);
-
-    _t0 = p.get<double>("G4Time");
-    _t0_sigma = p.get<double>("G4TimeJitter");
-    if(_t0_sigma < 0) this->abort("Cannot have a negative value for G4 time jitter");
-
-    _multi_min = p.get<size_t>("MultiMin");
-    _multi_max = p.get<size_t>("MultiMax");
-
-    _tpc_v  = p.get<std::vector<std::vector<unsigned short> > >("TPCRange");
-    auto const xrange = p.get<std::vector<double> > ("XRange");
-    auto const yrange = p.get<std::vector<double> > ("YRange");
-    auto const zrange = p.get<std::vector<double> > ("ZRange");
-
-    auto const part_cfg = p.get<fhicl::ParameterSet>("ParticleParameter");
-
-    _param_v.clear();
-    auto const pdg_v      = part_cfg.get<std::vector<std::vector<int>    > > ("PDGCode");
-    auto const minmult_v  = part_cfg.get<std::vector<unsigned short> > ("MinMulti");
-    auto const maxmult_v  = part_cfg.get<std::vector<unsigned short> > ("MaxMulti");
-    auto const weight_v   = part_cfg.get<std::vector<double> > ("ProbWeight");
-
-    auto kerange_v  = part_cfg.get<std::vector<std::vector<double> > > ("KERange");
-    auto momrange_v = part_cfg.get<std::vector<std::vector<double> > > ("MomRange");
-
-    if( (kerange_v.empty() && momrange_v.empty()) ||
-       (!kerange_v.empty() && !momrange_v.empty()) ) {
-        this->abort("Only one of KERange or MomRange must be empty!");
-    }
+  //
+  // Random engine initialization
+  //
+  fFlatRandom = std::make_unique<CLHEP::RandFlat>(fFlatEngine,0,1);
+  fNormalRandom = std::make_unique<CLHEP::RandGauss>(fFlatEngine);
+
+  produces< std::vector<simb::MCTruth>   >();
+  produces< sumdata::RunData, art::InRun >();
+
+  _debug = p.get<unsigned short>("DebugMode",0);
+  _revert = p.get<unsigned short>("Revert",0);
+  _use_boost = p.get<unsigned short>("UseBoost",0);
+  _t0 = p.get<double>("G4Time");
+  _t0_sigma = p.get<double>("G4TimeJitter");
+  if(_t0_sigma < 0) this->abort("Cannot have a negative value for G4 time jitter");
+
+  _multi_min = p.get<size_t>("MultiMin");
+  _multi_max = p.get<size_t>("MultiMax");
+
+  _tpc_v  = p.get<std::vector<std::vector<unsigned short> > >("TPCRange");
+  auto const xrange = p.get<std::vector<double> > ("XRange");
+  auto const yrange = p.get<std::vector<double> > ("YRange");
+  auto const zrange = p.get<std::vector<double> > ("ZRange");
+
+  auto const part_cfg = p.get<fhicl::ParameterSet>("ParticleParameter");
+  _gamma_beta_range = p.get<std::array<double,2> >("GammaBetaRange", {0.0,0.0});
```

Comment:
im confused if this stores velocity or gamma

Reply:
beta_gamma stores beta*gamma